### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by blankdots**
This PR extracts documentation from sources code, as well as updates part of the documentation with the new features and functionalities.

A `readthedocs.yml` was added for python3.6 build of docs (more info http://docs.readthedocs.io/en/latest/yaml-config.html).

One can preview the live docs at: http://localega.readthedocs.io/en/feature-code_docs/

Small typos fixed, hopefully none added.

### CODE REVIEW

File: docs/ingestion/db.rst [hunk 1]
```diff
@@ -24,5 +24,5 @@ in order to insert or manipulate the database entry.
                          from_user  errors.from_user%TYPE) RETURNS void
 
 
-Look at :doc:`the SQL definitions </../extras/db.sql>` if you are also
+Look at :doc:`the SQL definitions </../extras/db.sql>` if on is also
```
Comment (silverdaz) [inline docs/ingestion/db.rst]: eh?

File: docs/.gitignore [hunk 2]
```diff
@@ -3,3 +3,6 @@
 # =====================================
 _build/
 static/*.key
+
+# These should be generated every time
+lega.utils/
```
Comment (silverdaz) [inline docs/.gitignore]: Out of curiosity, who generates this folder?

File: lega/ingest.py [hunk 3]
```diff
@@ -2,23 +2,19 @@
...
-#
-# Re-Encryption Worker
-#
-####################################
+Re-Encryption Worker simply consumes message from the message queue.
 
-It simply consumes message from the message queue configured in the [worker] section of the configuration files.
-
-It defaults to the `tasks` queue.
+This is configured in the ``[worker]`` section of the configuration files.
```
Comment (silverdaz) [inline lega/ingest.py]: I think it's out of date.
I'm probably the right person to fix it 😛

File: lega/ingest.py [hunk 4]
```diff
@@ -2,23 +2,19 @@
...
 When a message is consumed, it must be of the form:
-* filepath
-* target
-* hash (of the unencrypted content)
-* hash_algo: the associated hash algorithm
+
+* ``filepath``
+* ``target``
+* ``hash`` (of the unencrypted content)
+* ``hash_algo`` - the associated hash algorithm
```
Comment (silverdaz) [inline lega/ingest.py]: We are missing the `stable_id`, and the content is anyway out of date.

File: lega/ingest.py [hunk 5]
```diff
@@ -43,12 +39,13 @@ def work(master_key, data):
...
 
     The data is of the form:
+
     * user id
     * a filepath
     * encrypted hash information (with both the hash value and the hash algorithm)
     * unencrypted hash information (with both the hash value and the hash algorithm)
 
-    The hash algorithm we support are MD5 and SHA256, for the moment.
+    .. note:: The hash algorithm we support are MD5 and SHA256, for the moment.
```
Comment (silverdaz) [inline lega/ingest.py]: The supported hash algorithm are, for the moment, MD5 and SHA256.

File: lega/utils/amqp.py [hunk 6]
```diff
@@ -1,3 +1,6 @@
+'''
+Insures communication with RabbitMQ Message Broker
```
Comment (silverdaz) [inline lega/utils/amqp.py]: Ensures

File: lega/vault.py [hunk 7]
```diff
@@ -2,20 +2,17 @@
...
-#
-# Listener moving files to the Vault
-#
-####################################
+Listener moving files to the Vault
 
 It simply consumes message from the message queue configured in the [vault] section.
 
-It defaults to the `completed` queue.
+It defaults to the ``completed`` queue.
```
Comment (silverdaz) [inline lega/vault.py]: Out of date information

File: lega/utils/amqp.py (hunk 6 above)
Comment (silverdaz) [reply lega/utils/amqp.py]: or...specify which insurance company it is. 😝

File: docs/ingestion/db.rst (hunk 1 above)
Comment (blankdots) [reply docs/ingestion/db.rst]: should be `one` ... typo, more academic than you :)

File: lega/ingest.py (hunk 3 above)
Comment (blankdots) [reply lega/ingest.py]: could you fix it ?

File: docs/.gitignore (hunk 2 above)
Comment (blankdots) [reply docs/.gitignore]: line 92 in docs/conf.py `autosummary_generate = True`

**Pull request merged**